File: app/src/main/java/com/example/healthtracker/feature/imc/Imc.java
package com.example.healthtracker.feature.imc;

import androidx.annotation.StringRes;

import com.example.healthtracker.common.base.BasePresenter;
import com.example.healthtracker.common.base.BaseView;
import com.example.healthtracker.model.CalcDao;

public interface Imc {

    interface View extends BaseView {
        void onRegisterImcValue();
    }

    interface Presenter extends BasePresenter {
        void registerImcValue(double imc, CalcDao dao);
        boolean validate(String height, String weight);
        double calculateImc(double height, double weight);
        @StringRes
        int getImcSituation(double imc);
    }

}

File: app/src/main/java/com/example/healthtracker/feature/imc/Imc.kt
package com.example.healthtracker.feature.imc

import androidx.annotation.StringRes
import com.example.healthtracker.model.CalcDao

interface Imc {

    interface View{
        fun onRegisterImcValue()
        fun displayFailure(message: String)
    }

    interface Presenter{
        fun registerImcValue(imc: Double, type: String, dao: CalcDao)
        fun validate(height: String, weight: String): Boolean
        fun calculateImc(height: Int, weight: Int): Double
        @StringRes
        fun getImcSituation(imc: Double): Int
    }


}